shunting-yard.cpp

```diff
@@ -691,6 +691,16 @@ packToken calculator::eval(TokenMap vars, bool keep_refs) const {
   }
 }
 
+std::unordered_set<std::string> calculator::get_variables() const {
+  std::unordered_set<std::string> ret;
+  for (const auto& i: RPN) {
+    if (i->type == tokType::VAR) {
+      ret.insert(static_cast<Token<std::string>*>(i)->val);
+    }
+  }
+  return ret;
+}
```

Comment:
Its not necessary to make these changes, but I think its better to hint at what this variable contains in its name:

```suggestion
std::unordered_set<std::string> calculator::get_variables() const {
  std::unordered_set<std::string> vars;
  for (const auto& i: RPN) {
    if (i->type == tokType::VAR) {
      vars.insert(static_cast<Token<std::string>*>(i)->val);
    }
  }
  return vars;
}
```